packages/@aws-cdk/aws-elasticloadbalancingv2/lib/nlb/network-listener.ts

```diff
@@ -84,16 +84,21 @@ export class NetworkListener extends BaseListener implements INetworkListener {
   private readonly certificates?: INetworkListenerCertificateProps[];
 
   constructor(scope: cdk.Construct, id: string, props: NetworkListenerProps) {
+    let proto = props.protocol || Protocol.Tcp;
+
+    if (props.certificates && props.certificates.length > 0) {
+      proto = Protocol.Tls;
```

Comment:
This is not the same behavior as what I described. If you now supply:

```ts
new NetworkListener(..., {
  protocol: Protocol.Tcp,
  certificates: [...]
});
```

The protocol will end up as TLS, but I want that to throw an error.

Also, I may not have said so explicitly enough in my previous review, but I'd prefer the rules in `validate()` to go into the constructor and `throw new Error('')`;.